dashboard/app/controllers/pd/workshop_enrollment_controller.rb

```diff
@@ -59,6 +59,18 @@ def new
         sign_in_url: CDO.studio_url("/users/sign_in?user_return_to=#{request.url}")
       }
 
+      # We only want to ask each signed-in teacher about demographics once a year.
+      # In this enrollment, we'll only ask if they haven't already submitted a
+      # teacher application for the current year (since it asks the same), and if
+      # this enrollment is for a local summer workshop (since this means it's for
+      # CSD/CSP, and they will only apply for one local summer workshop a year).
+      collect_demographics = !!current_user &&
+        Pd::Application::ApplicationBase.where(
+          user: current_user,
+          application_year: Pd::Application::ActiveApplicationModels::APPLICATION_CURRENT_YEAR
```

Comment:
I think we only care whether they have a teacher application, and if that is the case, a slightly simpler way to write this query would be
```Pd::Application::ActiveApplicationModels::TeacherApplicationClass.where(user: current_user).empty?```

If we expect facilitators to enroll in local summer workshops and we don't want to ask them these questions, then leaving it how you have is fine, because your current query won't ask these questions of people if we have a facilitator application for them.